elements/howto-accordion/howto-accordion.js

```diff
@@ -0,0 +1,533 @@
+        // all the animated children. Therefore, set all children to
+        // `z-index: 2` and set all the animated children to `z-index: 1` next.
+        child.style.zIndex = 2;
+      });
+
+      // Set `z-index: 1` on all animated children translate them to the
+      // start position. Because this function uses a CSS transition we don’t
+      // need to use `will-change`.
+      animatedChildren.forEach(child => {
+        child.style.position = 'relative';
+        child.style.zIndex = 1;
+        child.style.transform = `translateY(${startOffset}px)`;
+      });
+
+      // Wait two frames for all the styles to take effect.
+      return requestAnimationFramePromise()
+        .then(_ => requestAnimationFramePromise())
+        .then(_ => {
+          // Set up the CSS transition on all the children and set them to
+          // their end position.
+          animatedChildren.forEach(child => {
+            child.style.transform = `translateY(${endOffset}px)`;
+            child.classList.add('animating');
+          });
+          // Wait for the transition to end.
+          return transitionEndPromise(panel);
+        })
+        .then(_ => {
+          // Clean up all the temporary styles
+          animatedChildren.forEach(child => {
+            child.style.transform = '';
+            child.classList.remove('animating');
+          });
+          children.forEach(child => {
+            child.style.position = '';
+            child.style.zIndex = '';
+          });
+          this.style.overflow = '';
+          this.classList.remove('animating');
+        });
+    }
+  }
+  window.customElements.define('howto-accordion', HowtoAccordion);
+
+  /**
+   * `HowtoAccordionHeading` is the element for the headings in the accordion.
+   * Accordion to the WAI ARIA Best Practices, each heading needs to wrap a
+   * `<button>`. This element dispatches a `howto-accordion-toggle` event when
+   * it is supposed to expand.
+   *
+   * Click the button or pressing space or enter while the button has focus
+   * will expand the button. Changing the `expand` attribute or property will
+   * also cause the heading to expand.
+   */
+  class HowtoAccordionHeading extends HTMLElement {
+    // The element reacts to changes to the `expanded` attribute.
+    static get observedAttributes() {
+      return ['expanded'];
+    }
+
+    constructor() {
+      super();
+
+      // Binding event handlers to `this` ensures that `this` inside the event
+      // handler will always be the `<howto-accordion-heading>`, even the
+      // handler is hooked up to other elements.
+      this._onClick = this._onClick.bind(this);
+    }
+
+    /**
+     * `connectedCallback()` sets up the role, event handler and initial state.
+     */
+    connectedCallback() {
+      if(!this.hasAttribute('role'))
+        this.setAttribute('role', 'heading');
+      const button = this.querySelector('button');
+      if(!button)
```

Comment:
yeah if you used shadow dom you could either inject the button yourself or listen for it in a slotchange event. Since you always want a button, it seems like you'd just inject it yourself.